demosplan/DemosPlanCoreBundle/Logic/OzgKeycloakUserDataMapper.php

```diff
@@ -346,16 +353,23 @@ private function tryCreateNewUser(Orga $userOrga, array $requestedRoles): ?User
         }
 
         $userData = [
-            'lastname'      => $this->ozgKeycloakResponseValueObject->getVollerName(),
-            'email'         => $this->ozgKeycloakResponseValueObject->getEmailAdresse(),
-            'login'         => $this->ozgKeycloakResponseValueObject->getNutzerId(),
-            'gwId'          => $this->ozgKeycloakResponseValueObject->getProviderId(),
+            'firstname'      => $this->ozgKeycloakUserData->getFirstName(),
+            'lastname'      => $this->ozgKeycloakUserData->getLastName(),
+            'email'         => $this->ozgKeycloakUserData->getEmailAddress(),
+            'login'         => $this->ozgKeycloakUserData->getUserName(),
+            'gwId'          => $this->ozgKeycloakUserData->getUserId(),
             'customer'      => $this->customerService->getCurrentCustomer(),
             'organisation'  => $userOrga,
             'department'    => $this->getDepartmentToSetForUser($userOrga),
             'roles'         => $requestedRoles,
         ];
 
+        //in case of incoming first and last name, set this to the related fields of dplan-user.
```

Comment:
Why do you need this block? Does it do anything different than above?

Reply:
Leftover of rework of mapping "fullName" into lastName, instead just use the also available/incoming parts of firstName and lastName, which can be easily (1:!) mapped to dplan-user.